lib/HapticsManager.tsx

```diff
@@ -0,0 +1,32 @@
+import { atom, useAtom, useSetAtom } from "jotai"
+import { HapticEvent, HapticsFunctions, expoPlayHaptics } from "./Haptics"
+
+export const hapticsAtom = atom(true)
+
+export class HapticsManager implements HapticsFunctions {
+  play = async (hapticTrigger: HapticEvent) => {
+    const [areHapticsOn, setAreHapticsOn] = useAtom(hapticsAtom)
```

Comment:
Can't use atom in class, use atomWithStorage and set asyncstorage value in class instead